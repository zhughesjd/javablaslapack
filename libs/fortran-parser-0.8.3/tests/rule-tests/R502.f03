! Test declaration-type-spec, which is:
!      declaration-type-spec  is  intrinsic-type-spec (R403)
!                             or  TYPE ( derived-type-spec )
!                             or  CLASS ( derived-type-spec )
!                             or  CLASS ( * )
!
! Tested separately are: R403 and R455.
!
! declaration-type-spec is tested as part of a type-declaration-stmt (R501).
integer :: a          ! alt1
type(my_type) :: b    ! alt2
class(my_class) :: c  ! alt3
class(*) :: d         ! alt4

end

