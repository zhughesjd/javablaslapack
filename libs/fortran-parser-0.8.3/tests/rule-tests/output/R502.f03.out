*******************************************
args[1]: R502.f03
R1101:main-program__begin:
R 403:intrinsic-type-spec: keyword1=[@18,391:397='integer',<61>,10:0] keyword2=null type=400 hasKindSelector=false
R 502:declaration-type-spec: type=500
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@22,402:402='a',<209>,10:11]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 455:derived-type-spec: typeName=[@37,425:431='my_type',<209>,11:5] hasTypeParamSpecList=false
R 502:declaration-type-spec: type=501
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@42,437:437='b',<209>,11:17]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 455:derived-type-spec: typeName=[@51,455:462='my_class',<209>,12:6] hasTypeParamSpecList=false
R 502:declaration-type-spec: type=502
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@56,468:468='c',<209>,12:19]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 502:declaration-type-spec: type=503
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@68,490:490='d',<209>,13:12]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=4
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
